docs/server-side/v2/api-reference/analytics/track-events.mdx

```diff
@@ -0,0 +1,159 @@
+            "timestamp": "2023-02-23T10:20:52.911740569Z",
+            "type": "track.update.success",
+                "data": {
+                "room_id": "633a78ee3bea2af508356a34",
+                "session_id": "63f73bf05223403c9671c5c9",
+                "room_name": "exp",
+                "peer_id": "c8e85ab4-d533-4de0-ba7c-4c58a4de6c74",
+                "user_id": "187a1a92-150f-4506-83b7-d8a1cd716fb0",
+                "user_name": "John",
+                "joined_at": "2023-02-23T10:12:00.587243415Z",
+                "role": "host",
+                "track_id": "d3c7a05d-81ee-48a0-8a6a-ec36be966312",
+                "stream_id": "cc6934ca-0c91-4cb7-8a97-293be2d0d8a8",
+                "type": "audio",
+                "source": "regular",
+                "mute": false,
+                "started_at": "2023-02-23T10:12:00.781851067Z"
+            }
+        },
+        {
+            "version": "2.0",
+            "id": "f64ebeec-9ab2-4d36-b445-affe5cda90a0",
+            "timestamp": "2023-02-23T10:22:30.720047276Z",
+            "type": "track.remove.success",
+            "data": {
+                "room_id": "633a78ee3bea2af508356a34",
+                "session_id": "63f73bf05223403c9671c5c9",
+                "room_name": "exp",
+                "peer_id": "c8e85ab4-d533-4de0-ba7c-4c58a4de6c74",
+                "user_id": "187a1a92-150f-4506-83b7-d8a1cd716fb0",
+                "user_name": "John",
+                "joined_at": "2023-02-23T10:12:00.587243415Z",
+                "role": "host",
+                "track_id": "d3c7a05d-81ee-48a0-8a6a-ec36be966312",
+                "stream_id": "cc6934ca-0c91-4cb7-8a97-293be2d0d8a8",
+                "type": "audio",
+                "source": "regular",
+                "mute": false,
+                "started_at": "2023-02-23T10:12:00.781851067Z"
+                "stopped_at": "2023-02-23T10:22:30.718237635Z"
+            }
+        }
+    ]
+}
+```
+</ResponseBox>
+
+## Why would you use this API?
+- You can use this API to build tools around user activity which can answer questions like "How long a participant was unmuted?",
+"How long did the teacher turn on their video in the class?", "When did a presenter share their screen?"
+
+- To get the active duration of a peer, you can pass `peer_id`/`user_id` in the query parameter while fetching events, this would limit events only to specified peer/user.
+    - For each track identifiable with `track_id` you can check `type` and `source` properties in the event to find out if you want to consider this track.
+        (for example if you are looking for audio events, you can check for type: `audio`, source: `regular`)
+    - Find the sum of duration starting from the event with `mute: false` (Eunmute<sub>timestamp</sub>) to the next event with `mute: true`(Emute<sub>timestamp</sub>) until we reach `track.remove.success`.  <br />(i.e active duration: Σ Emute<sub>timestamp</sub> - Eunmute<sub>timestamp</sub>)
```

Comment:
personally find it hard to understand this. can we add pseudo code or the diagram?

Reply:
would be great if this was structured as 
* "calculating How long did the teacher turn on their video in the class?"
* "calculating When did a presenter share their screen?"

presenting these as examples with snippets for the specific question will make it easier to understand